### DISCUSSION TITLE
Many thanks

### DISCUSSION THREAD

**Discussion by jackshafer** [nickcoutsos/keymap-editor#117] [Q&A] [answered]
Nick, I didn't know where to drop you a not of thanks for your keymap editor, so I hope this works for you.

jackshafer

**Comment by nickcoutsos** [comment, marked as answer]
@jackshafer thanks so much for the kind words, I'm always excited to hear from people getting use out of the app. You can also find me on Discord :)

**Comment by pofixifop** [comment]
+❤️
I also am just newly dabbling my feet into the world of ZMK keyboards and... this app is saving my life. (Besides editing the default .keymap file directly, I also tried configuring miryoku 🤪) Thank you SO SO much for making this. I probably would have given up otherwise, or resigned to making only very simple &kp bindings. (no layers, combos, etc..)

**Comment by englmaxi** [comment]
@nickcoutsos thanks a lot from me too!! I love this tool, it makes editing my keymap fun.

**Comment by aexvir** [comment]
another huge thanks from my side, the experience of editing the keymap visually it's a gamechanger
much appreciated 🙂

**Comment by jakdept** [comment]
Seriously this is helpful, this is amazing.

I've been buried at work for a while, working on and off with this, and would've gotten the last bit to this, but also I really need something to check my syntax / fix my keymap, and this more interactive thing let me do that.

Amazing.

Have you PR'd this in some form to `zmkfirmware/zmk`? If not, like, this should be in the official docs imo, it's that helpful.

**Reply by nickcoutsos** [reply]
Hey, thanks very much for the kind words!

> Have you PR'd this in some form to `zmkfirmware/zmk`? If not, like, this should be in the official docs imo, it's that helpful.

No, I haven't. Even in the early stages of this project it wasn't really something you could put in a PR and say "here you go I got things started". The early version just barely worked (and for a time supported both QMK and ZMK which I don't think helped) and the entire development process has been one stubborn fool banging his head against understanding ZMK/Devicetree until useful software started taking shape.

I used to share a lot of progress updates in the ZMK discord server for a while and got a positive response from the community, however my impression is that while the ZMK team were generally not opposed to my posting about this project they were not interested in commenting on it either as a standalone project or a part of ZMK.

Best case scenario, ZMK gets a protocol for runtime keymap updates and I can try to start implementing on my own it while work on ZMK Studio gets underway. I would also not be terribly surprised if ZMK Studio arrives in some form at the same time as runtime updates and this becomes irrelevant in the blink of an eye.

In the meantime, this is mostly fun to work on and knowing how much it helps people really keeps me pushing it forward. So, again, thank you everyone for this!

**Comment by fabianpage** [comment]
Hei, thank you for this tool! It's very helpful for me in building and understanding my keymap :)

**Comment by alan0ford** [comment]
I'm stepped stone to offer my thanks... :-)

I'm a newbie in ZMK, i found your tool really useful - once configuration is set - to customize keymaps finely and quickly.

I had trouble understanding the .json file syntax to draw the layout correctly, but I managed copying from other projects 

(excuse me for my poor english)

**Comment by Scybin** [comment]
Here to offer my thanks. This tool makes updating ZMK keymaps so much easier. Your work is appreciated.
